lib/core.js

```diff
@@ -126,7 +126,12 @@ process.on('uncaughtException', err => {
 
 process.on('warning', warning => {
   const stack = impress.shortenStack(warning.stack);
-  impress.log.warn(`${warning.message} ${stack}`);
+  const message = `${warning.message} ${stack}`;
+  if (impress.log) {
+    impress.log.warn(message);
+  } else {
+    console.warn(message);
```

Comment:
Could you add a TODO here to note that this is just a fix to avoid a crash and not an actual solution to the problem. Which would be to move logger initialization to the very beginning of the impress and/or and some sort of buffer to it until it has fully initialized.